### PR TITLE
Deprecated `llvmlite.llvmpy`

### PR DESCRIPTION
**Pull request by apmasell**
Add deprecation warnings to `llvmlite.llvmpy` and all of its submodules.

### CODE REVIEW

**Review by sklam — COMMENTED** [review]
Can you add this deprecation to https://llvmlite.readthedocs.io/en/latest/user-guide/deprecation.html (the source is at https://github.com/numba/llvmlite/blob/main/docs/source/user-guide/deprecation.rst)?

File: docs/source/user-guide/deprecation.rst [hunk 1]
```diff
@@ -9,6 +9,57 @@ APIs that have become undesirable/obsolete. Any information about the schedule
...
+- replace `llvmlite.llvmpy.core.Builder.icmp` with
+  `llvmlite.ir.IRBuilder.icmp_signed` and `icmp_unsigned`, as appropriate
+- replace `llvmlite.llvmpy.core.Builder.fcmp` with
+  `llvmlite.ir.IRBuilder.fcmp_ordered` and `fcmp_unordered`, as appropriate
+- replace calls to the static methods of `llvmlite.llvmpy.core.Type` with the
+  constructors provided in `llvmlite.ir` (_e.g._, `Type.int(8)` with
+  `IntType(8)`)
+- Replace calls to the static methods of `llvmlite.llvmpy.core.Constant` with
+  calls to the constructor of `llvmlite.ir.Constant` or
+  `llvmlite.ir.Constanti.literal_struct`, as appropriate. Note that `stringz`
```
Comment (stuartarchibald) [inline docs/source/user-guide/deprecation.rst:52]:
```suggestion
  `llvmlite.ir.Constant.literal_struct`, as appropriate. Note that `stringz`
```

File: llvmlite/llvmpy/core.py [hunk 2]
```diff
@@ -3,6 +3,12 @@
 from llvmlite import ir
 from llvmlite import binding as llvm
 
+import warnings
+
+warnings.warn(
+    "The module `llvmlite.llvmpy.passes` is deprecated and will be removed in "
```
Comment (sklam) [inline llvmlite/llvmpy/core.py:9]:
```suggestion
    "The module `llvmlite.llvmpy.core` is deprecated and will be removed in "
```

**Review by sklam — APPROVED** [review]
LGTM

**Pull request merged**